src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -488,8 +490,14 @@ private static TypeSymbol MakeMergedTupleType(ArrayBuilder<DeconstructionVariabl
                 return null;
             }
 
-            return TupleTypeSymbol.Create(locationOpt: null, elementTypes: typesBuilder.ToImmutableAndFree(), elementLocations: default(ImmutableArray<Location>),
-                                    elementNames: default(ImmutableArray<string>), compilation: compilation, diagnostics: diagnostics);
+            return TupleTypeSymbol.Create(
+                locationOpt: null,
+                elementTypes: typesBuilder.ToImmutableAndFree(),
+                elementLocations: default(ImmutableArray<Location>),
+                elementNames: default(ImmutableArray<string>),
+                compilation: compilation,
+                diagnostics: diagnostics,
+                shouldCheckConstraints: false);
```

Comment:
This is called in the following case with element types `int` and `C`.
```cs
(int x, var (err1, y)) = (0, new C());
```
We already checked this for RHS in Binder_Expressions.cs and LHS (the deconstruction) will be checked after this point, for the whole `(int x, var (err1, y))` as we have precise locations for each element (x, err1 and y).